### PR TITLE
Correctly handle hyphens in search index

### PR DESCRIPTION
**Pull request by tomlev**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fix hyphen indexation on product search
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-4148
| How to test?  | create product with hyphen in its name, check the FO search header behavior. Try with multiple hyphens, space separated ones...

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9353)

### CODE REVIEW

File: classes/Search.php [hunk 1]
```diff
@@ -121,10 +150,10 @@ public static function sanitize($string, $id_lang, $indexation = false, $iso_cod
                 }
             }
             $string = implode(' ', $processed_words);
-            $string = preg_replace('/[._]+/', '', $string);
-            $string = ltrim(preg_replace('/([^ ])-/', '$1 ', ' '.$string));
-            $string = preg_replace('/[._]+/', '', $string);
-            $string = preg_replace('/[^\s]-+/', '', $string);
+            $string = str_replace(['.', '_'], '', $string);
+            if (!$keepHyphens) {
+                $string = ltrim(preg_replace('/([^ ])-/', '$1 ', ' ' . $string));
```
Comment (PierreRambaud) [inline classes/Search.php:153]: Why adding space first :/ You can just `ltrim($string, '- ');`

File: tests/Unit/Core/Business/Product/Search/SearchTest.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            'with space after hyphen'               => [
+                'input'    => 'test1- test2',
+                'langId'   => 1,
+                'expected' => ['test1', 'test2', 'test1-'],
+            ],
+            'with multiple space separated hyphens' => [
+                'input'    => 'test1 - - test2',
+                'langId'   => 1,
+                'expected' => ['test1', '-', 'test2'],
+            ],
```
Comment (PierreRambaud) [inline tests/Unit/Core/Business/Product/Search/SearchTest.php:116]: Missing tests:

- '  - - - - test test - - -testets ----'
- ' - - - - - - - - -test  '
- '                   test - - - - test ---------'

**Review by PierreRambaud — DISMISSED** [review]

**Comment by marionf** [comment]
@tomlev 

I have created 3 products: 

capture d ecran_77

If I search t-shirt or shirt I have no results

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @tomlev

**Comment by photoinvest** [comment]
Will this solution be in the next update?
Thank you for answering.

**Comment by khouloudbelguith** [comment]
Hi @photoinvest,

Yes, it is added in the 1.7.5.0 roadmap.

**Pull request merged**